internal/controller/template_controller_test.go

```diff
@@ -256,30 +256,33 @@ var _ = Describe("Template Controller", func() {
 				return nil
 			}).WithTimeout(timeout).WithPolling(interval).Should(Succeed())
 
-			By("Creating a management cluster object with proper required versions in status")
-			// must set status here since it's controller by another ctrl
-			mgmt := &hmcmirantiscomv1alpha1.Management{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: mgmtName,
-				},
-				Spec: hmcmirantiscomv1alpha1.ManagementSpec{
-					Release: "test-release",
-				},
-			}
-			Expect(k8sClient.Create(ctx, mgmt)).To(Succeed())
-			mgmt.Status = hmcmirantiscomv1alpha1.ManagementStatus{
-				AvailableProviders: []string{someProviderName, otherProviderName},
-				CAPIContracts:      map[string]hmcmirantiscomv1alpha1.CompatibilityContracts{someProviderName: {capiVersion: someExposedContract}, otherProviderName: {capiVersion: otherExposedContract}},
-			}
-			Expect(k8sClient.Status().Update(ctx, mgmt)).To(Succeed())
+			// By("Creating a management cluster object with proper required versions in status")
+			// // must set status here since it's controller by another ctrl
+			// mgmt := &hmcmirantiscomv1alpha1.Management{
+			// 	ObjectMeta: metav1.ObjectMeta{
+			// 		Name: mgmtName,
+			// 	},
+			// 	Spec: hmcmirantiscomv1alpha1.ManagementSpec{
+			// 		Release: "test-release",
+			// 	},
+			// }
+			// Expect(k8sClient.Create(ctx, mgmt)).To(Succeed())
+			// mgmt.Status = hmcmirantiscomv1alpha1.ManagementStatus{
+			// 	AvailableProviders: []string{someProviderName, otherProviderName},
+			// 	CAPIContracts:      map[string]hmcmirantiscomv1alpha1.CompatibilityContracts{someProviderName: {capiVersion: someExposedContract}, otherProviderName: {capiVersion: otherExposedContract}},
+			// }
+			// Expect(k8sClient.Status().Update(ctx, mgmt)).To(Succeed())
+			mgmt := &hmcmirantiscomv1alpha1.Management{}
+			key := client.ObjectKey{Name: mgmtName}
+			Expect(k8sClient.Get(ctx, key, mgmt)).To(Succeed())
 
 			By("Checking the management cluster appears")
 			Eventually(func() error {
 				if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(mgmt), mgmt); err != nil {
 					return err
 				}
 
-				if l := len(mgmt.Status.AvailableProviders); l != 2 {
+				if l := len(mgmt.Status.AvailableProviders); l != 3 {
```

Comment:
error has `2`, and at this line we have `3`. could you also elaborate why do we have to update the value? i mean, I see the three providers at the `suite_test.go:289` but at the moment of writing this comment, the reason is not clear. is it because the only instance of the `management` is to be used across all the tests (except the `mgmt-ctrl`) and at least one "expected"/"well-known provider" is expected? to me maybe it could also be beneficial to comment that explicitly

Reply:
> is it because the only instance of the management is to be used across all the tests
not in tests, but controllers expect management object with hardcoded name to exist.

However, I just ran the tests without adding provider to management object's status, and they pass. So I'll revert this change. Also, we create the object, then manually update the status and check whether the status is in expected state. Without running controller which may mutate the object, these checks make no sense.

Reply:
these checks are meant to ensure the objects have been actually updated and have no effect with further errors if any (e.g. `(1) create an object; (2) immediately check something; (3) object has been`, between `1` and `2` the `3` is not guaranteed hence the unexpected error might take place, hence it's better to have such checks)

Reply:
> these checks are meant to ensure the objects have been actually updated and have no effect with further errors if any (e.g. (1) create an object; (2) object has been created; (3) check something, between 1 and 3 the step 2 is not guaranteed hence the unexpected error might take place, hence it's better to have such checks)

it would make sense in case there were running reconcilers in background which might mutate the object. Currently, this checks only tests that the k8s client in this concrete test was able to interact with api-server running for test. In other words, these checks do not test our code, but k8s client.

Reply:
exactly. there are many checks of the same kind in the tests to avoid sporadic failures in the CI. better to have failures because of incorrect logic rather than having unrelated failures because of random whatever somewhere on the runners which caused sluggishness in a local running component